media/Config/buildDom.js

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// remove every options on DOM
+const emptyOptionBox = function(isImport) {
+  if (!isImport) {
+    const locaForSelect = document.querySelector('#locaForSelect');
+    while (locaForSelect.hasChildNodes()) {
+      locaForSelect.removeChild(locaForSelect.firstChild);
+    }
+    const toolList = document.querySelectorAll('.tools div');
+    for (let i = 0; i < toolList.length; i++) {
+      toolList[i].classList.remove('selected');
+    }
+  }
+  const optionsName = document.querySelector('#optionsName');
+  while (optionsName.hasChildNodes()) {
+    optionsName.removeChild(optionsName.firstChild);
+  }
+  const optionsValue = document.querySelector('#optionsValue');
+  while (optionsValue.hasChildNodes()) {
+    optionsValue.removeChild(optionsValue.firstChild);
+  }
+};
+
+// change tool name and change toggle button
+const changeCommonTags = function(tool) {
+  const h2Tag = document.querySelector('#toolName');
+  h2Tag.innerText = `Options for ${tool.type}`;
+  const tmpBtn = document.querySelector('#useBtn');
+  const useBtn = tmpBtn.cloneNode(true);
+  tmpBtn.parentNode.replaceChild(useBtn, tmpBtn);
+  if (tool.type.startsWith('one-import')) {
+    // you can find oneImportToggleFunction in buildImportDom.js
+    useBtn.addEventListener('click', oneImportToggleFunction);
+  } else {
+    useBtn.addEventListener('click', function() {
+      const optionFieldset = document.querySelector('#options');
+      if (tool.use === true) {
+        tool.use = false;
+        optionFieldset.disabled = true;
+      } else {
+        tool.use = true;
+        optionFieldset.disabled = false;
+      }
+      // you can find autoCompletePath in pathAutoComplete.js
+      autoCompletePath(tool);
+    });
+  }
+  const optionFieldset = document.querySelector('#options');
+  if (tool.type.startsWith('one-import')) {
+    if (oneImport.use === true) {
+      useBtn.checked = true;
+      optionFieldset.disabled = false;
+    } else {
+      useBtn.checked = false;
+      optionFieldset.disabled = true;
+    }
+  } else {
+    if (tool.use === true) {
+      useBtn.checked = true;
+      optionFieldset.disabled = false;
+    } else {
+      useBtn.checked = false;
+      optionFieldset.disabled = true;
+    }
```

Comment:
Inner `if - else - ` codes looks the same. can you extract them to lambda and call it?

Reply:
And this one.